pkg/base1/test-external.js

```diff
@@ -26,6 +26,38 @@ QUnit.test("external get", function (assert) {
     req.send();
 });
 
+QUnit.test("external fsread1", async assert => {
+    const done = assert.async();
+    assert.expect(5);
+    const resp = await cockpit.spawn(["bash", "-c", "size=$(stat --format '%s' /usr/lib/os-release); echo $size"]);
+    const filesize = resp.replace(/\n$/, "");
+
+    /* The query string used to open the channel */
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        path: '/usr/lib/os-release',
+        binary: "raw",
+        external: {
+            "content-disposition": 'attachment; filename="foo"',
+            "content-type": "application/octet-stream",
+        }
+    }));
+
+    const req = new XMLHttpRequest();
+    req.open("GET", "/cockpit/channel/" + cockpit.transport.csrf_token + '?' + query);
+    req.onreadystatechange = function() {
```

Comment:
Is there any chance I could convince you to write this async/await?

And what about reading the data bytes and making sure you get the same number as the header suggested?

Reply:
I guess that's a 'no' :)